File: frontend/src/types/agents.ts
export interface AgentMetrics {
  agent_id: string
  agent_name: string
  total_executions: number
  success_rate: number
  avg_duration: number
  last_execution: string | null
}

export interface AgentPerformance {
  agent_id: string
  agent_name: string
  total_executions: number
  successful_executions: number
  failed_executions: number
  success_rate: number
  avg_duration: number
  p95_duration: number
  p99_duration: number
  error_rate: number
  trends?: any[]
}

File: backend/src/models/schemas/agents.py
"""Agent schemas."""

from pydantic import BaseModel, Field
from typing import Optional, List, Dict
from datetime import datetime


class AgentMetrics(BaseModel):
    """Agent basic metrics."""

    agent_id: str
    agent_name: str
    total_executions: int = 0
    success_rate: float = 0.0
    avg_duration: float = 0.0
    last_execution: Optional[datetime] = None


class AgentPerformance(BaseModel):
    """Detailed agent performance metrics."""

    agent_id: str
    agent_name: str
    total_executions: int
    successful_executions: int
    failed_executions: int
    success_rate: float
    avg_duration: float
    p95_duration: float
    p99_duration: float
    error_rate: float
    trends: Optional[List[Dict]] = None


class AgentStats(BaseModel):
    """Statistical analysis for agent."""

    agent_id: str
    stats: Dict
    distribution: Optional[Dict] = None
    percentiles: Optional[Dict] = None
